### PR TITLE
[editor] Added commands for Open Symbol and Go To Line… and many more (fixes #217, #216)

### LINKED ISSUE #217: Go To Line
The go to line command shall use the widget from #212.

### PR DESCRIPTION
**Pull request by svenefftinge**
Signed-off-by: Sven Efftinge <sven.efftinge@typefox.io>

### CODE REVIEW

**Comment by kittaakos** [comment]
Let me quickly try this out locally.

**Comment by kittaakos** [comment]
You have removed the `Select All` menu item with your commit.
gh_616_missing_select_all


Additionally, and this might be unrelated but the quick command cannot be triggered with `Ctrl/Cmd`+`Shift`+`P` in FF.

**Comment by kittaakos** [comment]
Furthermore, I can see these in my log:
```
[2017-10-10T07:33:19.545Z]  WARN: Theia/49092 on Akoss-MBP.fritz.box:
    Collided keybinding is ignored;  [ [ { commandId: 'monaco.editor.action.goToImplementation',
          keyCode: [Object],
          accelerator: [Array] },
        ' collided with ',
        '[object Object]' ] ]
[2017-10-10T07:33:19.547Z]  WARN: Theia/49092 on Akoss-MBP.fritz.box:
    Collided keybinding is ignored;  [ [ { commandId: 'monaco.editor.action.toggleTabFocusMode',
          keyCode: [Object],
          accelerator: [Array] },
        ' collided with ',
        '[object Object]' ] ]
[2017-10-10T07:33:19.547Z]  WARN: Theia/49092 on Akoss-MBP.fritz.box:
    Collided keybinding is ignored;  [ [ { commandId: 'monaco.find.history.showNext',
          keyCode: [Object],
          accelerator: [Array] },
        ' collided with ',
        '[object Object]' ] ]
[2017-10-10T07:33:19.549Z]  WARN: Theia/49092 on Akoss-MBP.fritz.box:
    Collided keybinding is ignored;  [ [ { commandId: 'monaco.find.history.showPrevious',
          keyCode: [Object],
          accelerator: [Array] },
        ' collided with ',
        '[object Object]' ] ]
```

Even if that is a known issue, the log message does not help figuring out the what the problem is: `[object Object]`.

**Comment by svenefftinge** [comment]
> Additionally, and this might be unrelated but the quick command cannot be triggered with Ctrl/Cmd+Shift+P in FF.

Is unrelated.

**Comment by svenefftinge** [comment]
> Furthermore, I can see these in my log:
Is that in FF?

**Comment by kittaakos** [comment]
> Is that in FF?

I have tried in both FF and Chrome. I do not know where it comes from, so I will try again in both. Just a second...

**Comment by kittaakos** [comment]
I have received the above-mentioned key-binding collision messages in both Chrome and FF.

File: packages/core/src/browser/quick-open/quick-command-service.ts [hunk 1]
```diff
@@ -24,7 +24,8 @@ export class QuickCommandService implements QuickOpenModel {
     open(): void {
         // let's compute the items here to do it in the context of the currently activeElement
         this.items = [];
-        for (const command of this.commands.commands) {
+        const filteredAndSortedCommands = this.commands.commands.filter(a => a.label).sort((a, b) => a.label!.localeCompare(b.label!));
```
Comment (hexa00) [inline packages/core/src/browser/quick-open/quick-command-service.ts]: Forgive my laziness but what is `label!` here ? 
Also the sorting is done alphabetically now is that better than the fuzzy sort ?

Comment (svenefftinge) [reply packages/core/src/browser/quick-open/quick-command-service.ts]: I will rename it :)
Yes, it is better then fuzzy sort, which does sort based on the length of the label (and other unrelated things).

Comment (svenefftinge) [reply packages/core/src/browser/quick-open/quick-command-service.ts]: Ah, you mean the exclamation mark? It tells typescript to not worry about `undefined`. It doesn't statically understand that I did filter already before.

Comment (hexa00) [reply packages/core/src/browser/quick-open/quick-command-service.ts]: Yes... OK I had no idea thx.

**Comment by kittaakos** [comment]
> You have removed the Select All menu item with your commit.

👍 Fixed.

> Furthermore, I can see these in my log:

👍 I cannot see them anymore.

Got to line and symbol works good.

**Pull request merged**